src/Serializer/AbstractItemNormalizer.php

```diff
@@ -93,8 +101,12 @@ public function __construct(PropertyNameCollectionFactoryInterface $propertyName
     /**
      * {@inheritdoc}
      */
-    public function supportsNormalization($data, $format = null)
+    public function supportsNormalization($data, $format = null, array $context = [])
     {
+        if (!(($context[static::USE_API_PLATFORM] ?? false) || \in_array($format, self::FORMATS, true))) {
```

Comment:
just asking, but given that this method now relies on the context to answer, can we still systematically return `true` in the `hasCacheableSupportsMethod` ?

Reply:
We can't. And that's one of the reasons why this fix cannot work...

Reply:
how should we fix it ? :P

Reply:
Revert 😝

Reply:
Sometimes we gotta acknowledge we've made a wrong turn...

Reply:
Just return `false` in `hasCacheableSupportsMethod`.
@teohhanhui revert what? All our new input/output system? It doesn't look like a good idea. It's a very good move, and a long awaited one.

The proper fix for this specific problem is to use our own instance of the serializer instead of the global one provided by Symfony. To do so, we need to patch SF first, to provide convenient factories to wire and configure a serializer (basically, `FrameworkExtension` must be revamped). It will also benefit to Doctrine JSON ODM, and many other libraries.
But it's a hard work, and we need a quick fix first, such as this one. We'll improve later.